src/utils/daos_hdlr.c

```diff
@@ -1642,7 +1643,7 @@ parse_filename_dfs(const char *path, char **_obj_name, char **_cont_name)
 		}
 
 		if (strcmp(cont_name, ".") == 0) {
-			cont_name = strdup(cwd);
+			D_STRNDUP(cont_name, cwd, strlen(cwd) + 1);
```

Comment:
The length of the buffer is known here, so use it.  strlen() gives you no protection at all.
```suggestion
			D_STRNDUP(cont_name, cwd, 1024);
```